cpp/CMakeLists.txt

```diff
@@ -0,0 +1,13 @@
+
+
+set(SOURCE_FILES
+        EncryptionProvider.cpp
+        FileTransferAgent.cpp
+        SnowflakeS3Client.cpp
+        StorageClientFactory.cpp
+        )
+
+add_library(snowflakeclientcpp STATIC ${SOURCE_FILES})
+
+target_link_libraries(snowflakeclient)
```

Comment:
This looks like nop. The first argument is the target library followed by other dependencies.